img_merger.py

```diff
@@ -0,0 +1,229 @@
+import cv2
+import json
+import argparse
+from pathlib import Path
+from bs4 import BeautifulSoup
+
+from f2b import F2B, draw
+from cvat_utils.coco_utils import baseline_info_coco, f2b_to_coco
+from cvat_utils.xml_utils import baseline_info_xml, f2b_to_xml
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--f2b_file', help='path to file for f2b settings', default='cvat_images/originals/f2b_set.json', type=str)
+parser.add_argument('--cvat_annot_file', help='path to file for cvat annotations', default='cvat_images/originals/cvat_annot_coco.json', type=str)
+parser.add_argument('--biggie_annot_file', help='path to file for biggie annotations', default='cvat_images/originals/biggie_annot_coco.json', type=str)
+parser.add_argument('--vis_extension', help='extension for visualized smallies and annotated image output', default='jpg', type=str)
+args = parser.parse_args()
+
+f2b_filepath = Path(args.f2b_file)
+assert f2b_filepath.is_file(), 'img data file doesnt exist'
+with open(str(f2b_filepath)) as json_file:
+    f2b_data = json.load(json_file)
+
+biggie_annots_path = Path(args.biggie_annot_file)
+
+smol_annots_dir = f2b_filepath.parent / 'smol_annots'
+Path.mkdir(smol_annots_dir, exist_ok=True)
+
+cvat_annots_path = Path(args.cvat_annot_file)
+assert cvat_annots_path.is_file(), 'img data file doesnt exist'
+
+# check if 'coco' or 'cvat for images' format
+if cvat_annots_path.suffix == '.json':
+    assert biggie_annots_path.suffix=='.json', 'biggie_annot_file needs to be a json'
+
+    with open(str(cvat_annots_path)) as json_file:
+        cvat_annots = json.load(json_file)
+
+    biggie_annots = baseline_info_coco(cvat_annots)
+
+    # get category labels
+    cats_f2b = {}
+    for cat_dict in cvat_annots['categories']:
+        cats_f2b[cat_dict['id']] = cat_dict['name']
+
+    biggie_images = []
+    all_flatten_dets = []
+    all_smallie_annots_coco = []
+    f2b_merged_dicts = []
+    for i, img_name in enumerate(f2b_data):
+        # read original image and f2b settings
+        f2b_settings = f2b_data[img_name]
+
+        input_path = Path(f2b_settings["img_path"])
+        biggie = cv2.imread(str(input_path))
+
+        if biggie is None:
+            print(f'{input_path} does not exist')
+            continue
```

Comment:
the purpose of reading `biggie` here is (1) getting the img size (2) drawing the viz. However, (1) can be handled by storing it in the `f2b_settings.json`. and (2) can be made optional also. In this case, we are able to run this `img_merger.py` script even without access to the actual biggie data. 

I recommend that we do a `try` `except` here to see if biggie exists, if exist then we will viz, otherwise we won't viz. Then (1) can be retrieved from `f2b_settings.json`